src/main/java/com/internet/shop/dao/jdbc/UserDaoJdbcImpl.java

```diff
@@ -0,0 +1,194 @@
+package com.internet.shop.dao.jdbc;
+
+import com.internet.shop.dao.UserDao;
+import com.internet.shop.exceptions.DataProcessingException;
+import com.internet.shop.lib.Dao;
+import com.internet.shop.model.Role;
+import com.internet.shop.model.User;
+import com.internet.shop.util.ConnectionUtil;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+@Dao
+public class UserDaoJdbcImpl implements UserDao {
+    @Override
+    public Optional<User> findByLogin(String login) {
+        String query = "SELECT * FROM users WHERE login = ? AND deleted = FALSE";
+        Optional<User> searchedUser = Optional.empty();
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setString(1, login);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                searchedUser = Optional.of(composeExtractedUser(resultSet));
+            }
+            if (searchedUser.isEmpty()) {
+                return searchedUser;
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't find user by specified login: " + login, e);
+        }
+        searchedUser.get().setRoles(composeRoleSetByUserId(searchedUser.get().getId()));
+        return searchedUser;
+    }
+
+    @Override
+    public User create(User user) {
+        String query = "INSERT INTO users (user_name, login, password) VALUES (?, ?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection
+                    .prepareStatement(query, Statement.RETURN_GENERATED_KEYS);
+            statement.setString(1, user.getName());
+            statement.setString(2, user.getLogin());
+            statement.setString(3, user.getPassword());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                user.setId(resultSet.getLong(1));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't create provided user: " + user, e);
+        }
+        addRoles(user.getRoles(), user.getId());
+        return user;
```

Comment:
does user have roles_id null in create method, or they are initialized properly?

Reply:
returned user have null_id. to make returned user have role set with proper/initialized id I can use either 
- return(get(user.getId())) or
- user.setRoles(composeRoleSetByUserId(user.getId()));
  return user;
which one to choose?

Reply:
i think second option makes more sense

Reply:
chose second option

Reply:
done! I think i was uploading the changes at time you wrote your response) nice to know the prefered solution seemed more suitable for me as well